tokio/tests/sync_mpsc.rs

```diff
@@ -1017,4 +1017,81 @@ async fn test_tx_capacity() {
     assert_eq!(tx.max_capacity(), 10);
 }
 
+#[tokio::test]
+async fn test_rx_is_closed_when_calling_close_with_sender() {
+    // is_closed should return true after calling close but still has a sender
+    let (_tx, mut rx) = mpsc::channel::<()>(10);
+    rx.close();
+
+    assert!(rx.is_closed());
+}
+
+#[tokio::test]
+async fn test_rx_is_closed_when_dropping_all_senders() {
+    // is_closed should return true after dropping all senders
+    let (tx, mut rx) = mpsc::channel::<()>(10);
+    let another_tx = tx.clone();
+    let task = tokio::spawn(async move {
+        drop(another_tx);
+    });
+
+    drop(tx);
+    let _ = task.await;
+
+    assert!(rx.is_closed());
+}
+
+#[tokio::test]
+async fn test_rx_is_not_closed_when_there_are_senders() {
+    // is_closed should return false when there is a sender
+    let (_tx, mut rx) = mpsc::channel::<()>(10);
+    assert!(!rx.is_closed());
+}
+
+#[tokio::test]
+async fn test_rx_is_not_closed_when_there_are_senders_and_buffer_filled() {
+    // is_closed should return false when there is a sender, even if enough messages have been sent to fill the channel
+    let (tx, mut rx) = mpsc::channel(10);
+    for i in 0..10 {
+        assert!(tx.send(i).await.is_ok());
+    }
+    assert!(!rx.is_closed());
+}
+
+#[tokio::test]
+async fn test_rx_is_not_closed_when_there_are_messages_but_not_senders() {
+    // is_closed should return false when there is a permit (but no senders)
+    let (tx, mut rx) = mpsc::channel(10);
+    for i in 0..10 {
+        assert!(tx.send(i).await.is_ok());
+    }
+    drop(tx);
+    assert!(!rx.is_closed());
+}
+
+#[tokio::test]
+async fn test_rx_is_not_closed_when_there_are_messages_and_close_is_called() {
+    // is_closed should return false when there is a permit (but no senders)
+    let (tx, mut rx) = mpsc::channel(10);
+    for i in 0..10 {
+        assert!(tx.send(i).await.is_ok());
+    }
+    rx.close();
+    assert!(!rx.is_closed());
+}
+
+#[tokio::test]
+async fn test_rx_is_closed_after_consuming_messages() {
+    // is_closed should return false when there is a permit (but no senders)
+    let (tx, mut rx) = mpsc::channel(10);
+    for i in 0..10 {
+        assert!(tx.send(i).await.is_ok());
+    }
+    drop(tx);
+
+    assert!(!rx.is_closed());
+    while (rx.recv().await).is_some() {}
```

Comment:
Also here. It seems like there aren't any tests for the case where there is a permit but no sender.